data-products/src/main/scala/org/sunbird/analytics/util/TextbookUtils.scala

```diff
@@ -0,0 +1,42 @@
+package org.sunbird.analytics.util
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SQLContext
+import org.ekstep.analytics.framework.Params
+import org.ekstep.analytics.framework.util.HTTPClient
+import org.ekstep.analytics.model.report.ContentResult
+
+case class ContentInformation(id: String, ver: String, ts: String, params: Params, responseCode: String,result: TextbookResult)
+case class TextbookResult(count: Int, content: List[ContentResult])
+
+
+object TextbookUtils {
+
+  def getContentDataList(tenantId: String, restUtil: HTTPClient)(implicit sc: SparkContext): TextbookResult = {
+    implicit val sqlContext = new SQLContext(sc)
+    val url = Constants.COMPOSITE_SEARCH_URL
+    //    println("tenantId: " + tenantId)
+    val request = s"""{
+                     |      "request": {
+                     |        "filters": {
+                     |           "status": ["Live","Draft","Review","Unlisted"],
+                     |          "contentType": ["Resource"],
+                     |          "createdFor": "$tenantId"
+                     |        },
+                     |        "fields": ["channel","identifier","board","gradeLevel",
+                     |          "medium","subject","status","creator","lastPublishedOn","createdFor",
+                     |          "createdOn","pkgVersion","contentType",
+                     |          "mimeType","resourceType", "lastSubmittedOn"
+                     |        ],
+                     |        "limit": 10000,
+                     |        "facets": [
+                     |          "status"
+                     |        ]
+                     |      }
+                     |    }""".stripMargin
+
+    val resultData = restUtil.post[ContentInformation](url, request)
+      val finalResult = resultData.result
+    finalResult
```

Comment:
Can we build a retry logic? Why don't we use Unirest here which would internally automatically retry for configured number of times. Reach out to @manju, he has used it in one samza job

Reply:
Done